src/utilities.h

```diff
@@ -432,6 +448,57 @@ bool isStandardPrefixName(const std::string &name);
  */
 size_t getVariableIndexInComponent(const ComponentPtr &component, const VariablePtr &variable);
 
+/**
+ * @brief Test to determine if @p entity1 is a child of @p entity2.
+ *
+ * Test to see if @p entity1 is a child of @p entity2.  Returns @c true if
+ * @p entity1 is a child of @p entity2 and @c false otherwise.
+ *
+ * @param entity1 The @c Entity to test if it is a child of @p entity2.
+ * @param entity2 The @c Entity that is potentially the parent of @p entity1.
+ *
+ * @return @c true if @p entity1 is a child of @p entity2 and @c false otherwise.
+ */
+bool isEntityChildOf(const EntityPtr &entity1, const EntityPtr &entity2);
+
+/**
+ * @brief Test to determine if @p entity1 and @p entity2 are siblings.
+ *
+ * Test to determine if @p entity1 and @p entity2 are siblings.  Returns
+ * @c true if @p entity1 and @p entity2 are siblings, @c false otherwise.
+ *
+ * @param entity1 An @c Entity to test if it is a sibling to @p entity2.
+ * @param entity2 An @c Entity to test if it is a sibling to @p entity1.
+ *
+ * @return @c true if @p entity1 and @p entity2 are siblings, @c false otherwise.
+ */
+bool areEntitiesSiblings(const EntityPtr &entity1, const EntityPtr &entity2);
+
+/**
+ * @brief Determine the interface type of the @p variable.
+ *
+ * Determines the interface type of the given @p variable.  If the variable
+ * type is determined to be NONE this indicates an error (if the variable has
+ * at least one equivalent variable).
+ *
```

Comment:
For variables with at least one equivalent variable, returning an **interface** type of NONE indicates an error.